### PR TITLE
Userland: POSIX compliance improvements

### PR DESCRIPTION
**Pull request by boricj**
Fix various stuff I've encountered while attempting to bootstrap pkgsrc on SerenityOS. It's not working yet, at the very least `expr` still needs a POSIX basic regular expression engine for compliance (#8506).

### CODE REVIEW

**Comment by bgianfo** [comment]
Minor nit pick... none of the commit messages mention the motivation for any of these changes?

File: Userland/Utilities/id.cpp [hunk 1]
```diff
@@ -4,14 +4,15 @@
...
+#include <LibCore/Account.h>
 #include <LibCore/ArgsParser.h>
 #include <alloca.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdio.h>
 #include <unistd.h>
 
-static int print_id_objects();
+static int print_id_objects(const Core::Account&);
```
Comment (gunnarbeutner) [inline Userland/Utilities/id.cpp:16]: This should be east const. :)

**Review by MaxWipfli — APPROVED** [review]
Just a couple small comments. Seems very sensible to use `Core::Account` for `id(1)`.

File: Userland/Utilities/id.cpp [hunk 2]
```diff
@@ -58,8 +61,27 @@ int main(int argc, char** argv)
...
+        } else {
+            auto result = Core::Account::from_name(user_str, false);
+            if (result.is_error()) {
+                warnln("Couldn't retrieve user name '{}': {}", user_str, result.error());
+                return 1;
+            }
+            account = result.release_value();
+        }
+    } else
+        account = Core::Account::self(false);
```
[the remark this answers is not in the document]
Comment (awesomekling) [reply Userland/Utilities/id.cpp]: The project coding style does indeed say to have symmetry in curlies :)

**Pull request merged**